src/sources/datadog/agent.rs

```diff
@@ -10,19 +10,12 @@ use crate::{
 };
 use bytes::Bytes;
 use chrono::Utc;
-use lazy_static::lazy_static;
 use regex::Regex;
 use serde::{Deserialize, Serialize};
 use std::{collections::HashMap, net::SocketAddr, sync::Arc};
 use vector_core::event::LogEvent;
-use warp::http::StatusCode;
-
 use warp::http::HeaderMap;
-
-lazy_static! {
```

Comment:
Removing `lazy_static`s seems to be a common theme with your perf PRs - what's the issue with them?

Reply:
So the problem with `lazy_static` for vector is that what's actually static here -- in the sense of appearing in the static portion of the executable -- is a mutable place in memory for putting a value into, potentially with a race between multiple threads. Writing and read, then, implies an atomic barrier has to be passed through which on x86 will cause threads to queue. 

Used to be lazy_static was the only game in town. Now we should prefer the use of once_cell where we _do_ actually need a globalish variable or, better, just put the data local to whatever struct needs to make use of it.

Reply:
Thanks!